src/lib.rs

```diff
@@ -139,6 +139,17 @@ pub(crate) const CRINGE_ESLINT_DISABLE: &str = "/* eslint-disable */
 // #[cfg(doctest)]
 // doc_comment::doctest!("../README.md");
 
+/// The configuration for the generator
+#[derive(Default)]
+pub struct ExportConfiguration {
+    /// The name of the plugin to invoke.
+    ///
+    /// If there is no plugin name (i.e. this is an app), this should be `None`.
+    pub(crate) plugin_name: Option<String>,
+    /// The specta export configuration
+    pub(crate) spectra_config: SpectaExportConfiguration,
```

Comment:
The `Cow` is an easy way of getting an enum of `String` or `&'static str`. Being able to store `&'static str` is nice because it doesn't require the heap allocation like `String` does (hence why Rust has two string types).

This is also what Specta v2 does internally for all strings so consistency is also nice.